bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/CommunicationManager.java

```diff
@@ -427,118 +423,37 @@ private <T extends Type> void handleEvent(String itemName, T type, @Nullable Str
     @SuppressWarnings("unchecked")
     private <T extends Type> @Nullable T toAcceptedType(T originalType, Channel channel,
             Function<@Nullable String, @Nullable List<Class<? extends T>>> acceptedTypesFunction, Item item) {
-        String acceptedItemType = channel.getAcceptedItemType();
-
-        // DecimalType command sent to a NumberItem with dimension defined:
-        if (originalType instanceof DecimalType type && hasDimension(item, acceptedItemType)) {
-            @Nullable
-            QuantityType<?> quantityType = convertToQuantityType(type, item, acceptedItemType);
-            if (quantityType != null) {
-                return (T) quantityType;
-            }
-        }
+        String channelAcceptedItemType = channel.getAcceptedItemType();
```

Comment:
Yes, I would agree - this feels too early here - adapting to the channel type must only happen after the profile has been applied. But this can probably be tackled in a separate PR, since it is like this since the beginning.